tools/cli/commands/build.js

```diff
@@ -825,10 +827,15 @@ async function buildProject( t ) {
 		for ( const key of depTypes ) {
 			if ( packageJson[ key ] ) {
 				for ( const [ pkg, ver ] of Object.entries( packageJson[ key ] ) ) {
-					if ( ver.startsWith( 'workspace:* || ' ) ) {
-						packageJson[ key ][ pkg ] = ver.substring( 15 );
-					} else if ( ver === 'workspace:*' ) {
-						delete packageJson[ key ][ pkg ];
+					if ( ver === 'workspace:*' ) {
+						for ( const ctxPkg of Object.values( t.ctx.versions ) ) {
+							if ( ctxPkg.name === pkg ) {
+								let massagedVer = ctxPkg.version;
+								massagedVer = `^${ massagedVer }`;
```

Comment:
Could you please explain why we are truncating the PHP ones? I guess I thought it was about providing a wider resolution set, but I think I'm missing the actual reasoning here.

Reply:
The main reason I think is "because we always have". I can't find any good articles about two-component versus three-component deps.

In theory a patch-level version bump should only contain bugfixes, so most of the time `^1.2` doesn't really say much more than `^1.2.3` as to what's actually needed to use the package. And it might make an upgrade slightly easier if someone can wait on updating from `^1.2.0`. On the other hand, since a patch bump should only have bugfixes, it shouldn't be that hard for someone to just go to 1.2.3 anyway. 🤷

Personally I'd be happy to get rid of the truncation for PHP packages too, simplifies our code here and better indicates what we actually tested with,

Reply:
> Personally I'd be happy to get rid of the truncation for PHP packages too

[ebf7104](https://github.com/Automattic/jetpack/pull/27043/commits/ebf710412b7a3b695da4dddb5b458f877aa65b6e)